In [2]:
import pandas as pd
import numpy as np

# *Initial Inspection*

## *Closed Deals Dataset*

In [3]:
closed_deals = pd.read_csv("../data/olist_closed_deals_dataset.csv")
closed_deals.head()

,mql_id,seller_id,sdr_id,sr_id,won_date,business_segment,lead_type,lead_behaviour_profile,has_company,has_gtin,average_stock,business_type,declared_product_catalog_size,declared_monthly_revenue
0,5420aad7fec3549a85876ba1c529bd84,2c43fb513632d29b3b58df74816f1b06,a8387c01a09e99ce014107505b92388c,4ef15afb4b2723d8f3d81e51ec7afefe,26-02-2018 19:58,pet,online_medium,cat,NaN,NaN,NaN,reseller,NaN,0
1,a555fb36b9368110ede0f043dfc3b9a0,bbb7d7893a450660432ea6652310ebb7,09285259593c61296eef10c734121d5b,d3d1e91a157ea7f90548eef82f1955e3,08-05-2018 20:17,car_accessories,industry,eagle,NaN,NaN,NaN,reseller,NaN,0
2,327174d3648a2d047e8940d7d15204ca,612170e34b97004b3ba37eae81836b4c,b90f87164b5f8c2cfa5c8572834dbe3f,6565aa9ce3178a5caf6171827af3a9ba,05-06-2018 17:27,home_appliances,online_big,cat,NaN,NaN,NaN,reseller,NaN,0
3,f5fee8f7da74f4887f5bcae2bafb6dd6,21e1781e36faf92725dde4730a88ca0f,56bf83c4bb35763a51c2baab501b4c67,d3d1e91a157ea7f90548eef82f1955e3,17-01-2018 13:51,food_drink,online_small,NaN,NaN,NaN,NaN,reseller,NaN,0
4,ffe640179b554e295c167a2f6be528e0,ed8cb7b190ceb6067227478e48cf8dde,4b339f9567d060bcea4f5136b9f5949e,d3d1e91a157ea7f90548eef82f1955e3,03-07-2018 20:17,home_appliances,industry,wolf,NaN,NaN,NaN,manufacturer,NaN,0


In [4]:
closed_deals.info()

<class 'pandas.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    str    
 1   seller_id                      842 non-null    str    
 2   sdr_id                         842 non-null    str    
 3   sr_id                          842 non-null    str    
 4   won_date                       842 non-null    str    
 5   business_segment               841 non-null    str    
 6   lead_type                      836 non-null    str    
 7   lead_behaviour_profile         665 non-null    str    
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     str    
 11  business_type                  832 non-null    str    
 12  declared_product_catalog_size  69 non-null     float64
 13  d

In [5]:
closed_deals["has_gtin"].value_counts(dropna=False)

has_gtin
NaN      778
True      54
False     10
Name: count, dtype: int64

In [6]:
closed_deals["won_date"] = pd.to_datetime(closed_deals["won_date"])


C:\Users\Aarushi Sharma\AppData\Local\Temp\ipykernel_17752\2500625901.py:1: UserWarning: Parsing dates in %d-%m-%Y %H:%M format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  closed_deals["won_date"] = pd.to_datetime(closed_deals["won_date"])


In [7]:
closed_deals.columns

Index(['mql_id', 'seller_id', 'sdr_id', 'sr_id', 'won_date',
       'business_segment', 'lead_type', 'lead_behaviour_profile',
       'has_company', 'has_gtin', 'average_stock', 'business_type',
       'declared_product_catalog_size', 'declared_monthly_revenue'],
      dtype='str')

In [8]:
closed_deals.info()

<class 'pandas.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   mql_id                         842 non-null    str           
 1   seller_id                      842 non-null    str           
 2   sdr_id                         842 non-null    str           
 3   sr_id                          842 non-null    str           
 4   won_date                       842 non-null    datetime64[us]
 5   business_segment               841 non-null    str           
 6   lead_type                      836 non-null    str           
 7   lead_behaviour_profile         665 non-null    str           
 8   has_company                    63 non-null     object        
 9   has_gtin                       64 non-null     object        
 10  average_stock                  66 non-null     str           
 11  business_type                 

In [9]:
closed_deals.isnull().sum()

mql_id                             0
seller_id                          0
sdr_id                             0
sr_id                              0
won_date                           0
business_segment                   1
lead_type                          6
lead_behaviour_profile           177
has_company                      779
has_gtin                         778
average_stock                    776
business_type                     10
declared_product_catalog_size    773
declared_monthly_revenue           0
dtype: int64

In [10]:
closed_deals.duplicated().sum()

np.int64(0)

In [11]:
closed_deals['won_date'].min()

Timestamp('2017-12-05 02:00:00')

In [12]:
closed_deals['won_date'].max()

Timestamp('2018-11-14 18:04:00')

In [13]:
closed_deals["mql_id"].duplicated().sum()

np.int64(0)

In [14]:

closed_deals["seller_id"].duplicated().sum()

np.int64(0)

In [15]:

closed_deals["sdr_id"].duplicated().sum()

np.int64(810)

In [16]:

closed_deals["sr_id"].duplicated().sum()

np.int64(820)

In [17]:
closed_deals["business_segment"].value_counts()

business_segment
home_decor                         105
health_beauty                       93
car_accessories                     77
household_utilities                 71
construction_tools_house_garden     69
audio_video_electronics             64
computers                           34
pet                                 30
food_supplement                     28
food_drink                          26
sports_leisure                      25
bed_bath_table                      22
bags_backpacks                      22
toys                                20
fashion_accessories                 19
home_office_furniture               14
stationery                          13
phone_mobile                        13
small_appliances                    12
handcrafted                         12
baby                                10
music_instruments                    9
books                                9
watches                              8
jewerly                              8
home_app

In [18]:

closed_deals["lead_type"].value_counts()

lead_type
online_medium      332
online_big         126
industry           123
offline            104
online_small        77
online_beginner     57
online_top          14
other                3
Name: count, dtype: int64

In [19]:

closed_deals["business_type"].value_counts()

business_type
reseller        587
manufacturer    242
other             3
Name: count, dtype: int64

In [20]:
closed_deals["business_segment"].value_counts(dropna=False)

business_segment
home_decor                         105
health_beauty                       93
car_accessories                     77
household_utilities                 71
construction_tools_house_garden     69
audio_video_electronics             64
computers                           34
pet                                 30
food_supplement                     28
food_drink                          26
sports_leisure                      25
bed_bath_table                      22
bags_backpacks                      22
toys                                20
fashion_accessories                 19
home_office_furniture               14
stationery                          13
phone_mobile                        13
small_appliances                    12
handcrafted                         12
baby                                10
music_instruments                    9
books                                9
watches                              8
jewerly                              8
home_app

In [21]:
(closed_deals.isnull().sum() / len(closed_deals) * 100).sort_values(ascending=False)

has_company                      92.517815
has_gtin                         92.399050
average_stock                    92.161520
declared_product_catalog_size    91.805226
lead_behaviour_profile           21.021378
business_type                     1.187648
lead_type                         0.712589
business_segment                  0.118765
won_date                          0.000000
seller_id                         0.000000
mql_id                            0.000000
sr_id                             0.000000
sdr_id                            0.000000
declared_monthly_revenue          0.000000
dtype: float64

In [22]:
closed_deals['lead_behaviour_profile'].value_counts(dropna=False)

lead_behaviour_profile
cat            407
NaN            177
eagle          123
wolf            95
shark           24
cat, wolf        8
eagle, wolf      3
eagle, cat       3
shark, cat       1
shark, wolf      1
Name: count, dtype: int64

In [23]:
closed_deals['lead_behaviour_profile'].isnull().sum()

np.int64(177)

In [24]:
closed_deals['has_company'].value_counts()

has_company
True     58
False     5
Name: count, dtype: int64

In [25]:
closed_deals['has_gtin'].value_counts()

has_gtin
True     54
False    10
Name: count, dtype: int64

In [26]:
filtered = closed_deals[closed_deals['has_gtin'].isnull() == False]
filtered

,mql_id,seller_id,sdr_id,sr_id,won_date,business_segment,lead_type,lead_behaviour_profile,has_company,has_gtin,average_stock,business_type,declared_product_catalog_size,declared_monthly_revenue
30,4bb1ec1ff58ad39558e8468054bfc469,4fae87d32467e18eb46e4a76a0a0b9ce,9d12ef1a7eca3ec58c545c678af7869c,a8387c01a09e99ce014107505b92388c,2018-07-19 18:06:00,small_appliances,online_medium,cat,True,True,20-50,reseller,NaN,0
37,a8d2884db49e0769ad4a0b8dce2e143c,d65f74c704563bcb8baa915fbf8cfa1e,6aa3b86a83d784b05f0e37e26b20860d,de63de0d10a6012430098db33c679b0b,2018-09-27 12:31:00,health_beauty,online_medium,cat,NaN,False,NaN,reseller,15.0,20000
41,20fcec0873b39b4df3df34140d77d6e7,cee3c487c56d29554a22d112af618301,084b6fbb10729ed4da8c3d3f5a3ae7c9,0a0fb2b07d841f84fb6714e35c723075,2018-11-14 18:04:00,other,other,cat,False,False,01-May,other,120.0,6000
70,58caf27a48e7930aefd5437298d64a70,922f6795c9e65f2485d4e63af76586ff,45749fb708130f78d0db07d8d80f030b,34d40cdaf94010a1d05b0d6212f9e909,2018-09-21 18:18:00,bed_bath_table,online_medium,"cat, wolf",True,True,May-20,reseller,50.0,30000
86,4ddc45a9de1065695a90e87be37471c0,9d590e1df0faffd9ceaa1f0f0ea9af2f,e7dff61b78bebffa71678e126ce669ad,2695de1affa7750089c0455f8ce27021,2018-09-21 18:49:00,audio_video_electronics,online_medium,wolf,True,True,May-20,reseller,50.0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
780,7720c6892d447539b71d8bd9e4d1ef60,3786da2e9afd5d55617521bd1685dec0,f42a2bd194f7802ab052a815c8de65b7,a8387c01a09e99ce014107505b92388c,2018-10-01 17:47:00,toys,online_medium,"eagle, cat",True,True,unknown,reseller,300.0,25000
787,a57bf18e19b6b17e3d4fbc20561e2055,2a17b229ca84523393b003b7e0bfae63,f42a2bd194f7802ab052a815c8de65b7,9d12ef1a7eca3ec58c545c678af7869c,2018-08-20 14:41:00,toys,online_small,"eagle, cat",True,False,unknown,manufacturer,100.0,50000
810,d5cb61eef98bd237e41bc2225263f823,7e1f0755f1c75e301dfa37c21fd01efe,e4a6222cdb5b34375400904f03d8e6a5,34d40cdaf94010a1d05b0d6212f9e909,2018-11-12 19:17:00,other,other,"eagle, wolf",True,True,May-20,other,305.0,120000
811,9c2bfd97ce00ce9553bdc0f34c3532b6,ae9cf62cf691b753d4f0d3d40731cc3c,370c9f455f93a9a96cbe9bea48e70033,de63de0d10a6012430098db33c679b0b,2018-09-10 20:48:00,health_beauty,online_medium,shark,True,True,50-200,reseller,300.0,0


## *MQL Leads Dataset*

In [27]:
qualified_leads = pd.read_csv("../data/olist_marketing_qualified_leads_dataset.csv")
qualified_leads.head()

,mql_id,first_contact_date,landing_page_id,origin
0,dac32acd4db4c29c230538b72f8dd87d,2018-02-01,88740e65d5d6b056e0cda098e1ea6313,social
1,8c18d1de7f67e60dbd64e3c07d7e9d5d,2017-10-20,007f9098284a86ee80ddeb25d53e0af8,paid_search
2,b4bc852d233dfefc5131f593b538befa,2018-03-22,a7982125ff7aa3b2054c6e44f9d28522,organic_search
3,6be030b81c75970747525b843c1ef4f8,2018-01-22,d45d558f0daeecf3cccdffe3c59684aa,email
4,5420aad7fec3549a85876ba1c529bd84,2018-02-21,b48ec5f3b04e9068441002a19df93c6c,organic_search


In [28]:
qualified_leads.info()

<class 'pandas.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   mql_id              8000 non-null   str  
 1   first_contact_date  8000 non-null   str  
 2   landing_page_id     8000 non-null   str  
 3   origin              7940 non-null   str  
dtypes: str(4)
memory usage: 250.1 KB


In [29]:
qualified_leads.columns

Index(['mql_id', 'first_contact_date', 'landing_page_id', 'origin'], dtype='str')

In [30]:
qualified_leads['first_contact_date'] = pd.to_datetime(qualified_leads['first_contact_date'])

In [31]:
qualified_leads.info()

<class 'pandas.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   mql_id              8000 non-null   str           
 1   first_contact_date  8000 non-null   datetime64[us]
 2   landing_page_id     8000 non-null   str           
 3   origin              7940 non-null   str           
dtypes: datetime64[us](1), str(3)
memory usage: 250.1 KB


In [32]:
qualified_leads.describe()

,first_contact_date
count,8000
mean,2018-02-05 14:19:51.600000
min,2017-06-14 00:00:00
25%,2017-12-31 00:00:00
50%,2018-02-25 00:00:00
75%,2018-04-15 06:00:00
max,2018-05-31 00:00:00


In [33]:
qualified_leads.isnull().sum()

mql_id                 0
first_contact_date     0
landing_page_id        0
origin                60
dtype: int64

In [34]:
qualified_leads['origin'].value_counts()

origin
organic_search       2296
paid_search          1586
social               1350
unknown              1099
direct_traffic        499
email                 493
referral              284
other                 150
display               118
other_publicities      65
Name: count, dtype: int64

In [35]:
qualified_leads['origin'].value_counts(dropna=False)

origin
organic_search       2296
paid_search          1586
social               1350
unknown              1099
direct_traffic        499
email                 493
referral              284
other                 150
display               118
other_publicities      65
NaN                    60
Name: count, dtype: int64

In [36]:
qualified_leads['landing_page_id'].value_counts()

landing_page_id
b76ef37428e6799c421989521c0e5077    912
22c29808c4f815213303f8933030604c    883
58326e62183c14b0c03085c33b9fdc44    495
88740e65d5d6b056e0cda098e1ea6313    445
ce1a65abd0973638f1c887a6efcfa82d    394
                                   ... 
5215a2f4c4cb554cb18c8de3b4f06df5      1
d27094ded8d5242d4d2ba9d7e5f176b4      1
ffc62d6945fd1423539ec6d96326c37c      1
6ec35d2451cfa0e2e1c8595ec4ad099d      1
3683ba5726e741af4b8b02bcaef4e7b3      1
Name: count, Length: 495, dtype: int64

In [37]:
qualified_leads['landing_page_id'].duplicated().sum()

np.int64(7505)

In [38]:
qualified_leads['mql_id'].duplicated().sum()

np.int64(0)

## KPIs

#### Lead Conversion Rate 
- ((closed deals/total MQL's) * 100) 
- (842/8000) * 100 
- **10.53%**


#### Num0ber of MQLs 
- **800**


#### Channel Performance
- organic_search       2296
- paid_search          1586
- social               1350
- unknown              1099
- direct_traffic        499
- email                 493
- referral              284
- other                 150
- display               118
- other_publicities      65
- NaN                    60


#### Funnel Drop-off % 
- (((total MQL's - closed deals)/ total MQL's) * 100)
- (((8000 - 842)/8000) * 100)
- **89.48%**